app/src/test/kotlin/fr/free/nrw/commons/nearby/NearbyParentFragmentPresenterTest.kt

```diff
@@ -0,0 +1,320 @@
+package fr.free.nrw.commons.nearby
+
+import com.nhaarman.mockitokotlin2.*
+import fr.free.nrw.commons.bookmarks.locations.BookmarkLocationsDao
+import fr.free.nrw.commons.location.LatLng
+import fr.free.nrw.commons.location.LocationServiceManager
+import fr.free.nrw.commons.nearby.contract.NearbyParentFragmentContract
+import fr.free.nrw.commons.nearby.presenter.NearbyParentFragmentPresenter
+import org.junit.Before
+import org.junit.Test
+import org.junit.Assert.*
+import org.mockito.Mock
+import org.mockito.Mockito
+import org.mockito.MockitoAnnotations
+
+
+/**
+ * The unit test class for NearbyParentFragmentPresenter
+ */
+class NearbyParentFragmentPresenterTest {
+    @Mock
+    internal lateinit var nearbyParentFragmentView: NearbyParentFragmentContract.View
+    @Mock
+    internal lateinit var bookmarkLocationsDao: BookmarkLocationsDao
+    @Mock
+    internal lateinit var latestLocation: LatLng
+    @Mock
+    internal lateinit var cameraTarget: LatLng
+    @Mock
+    internal lateinit var selectedLabels: List<Label>
+
+    private lateinit var nearbyPresenter: NearbyParentFragmentPresenter
+    private lateinit var latestLocationSpy: LatLng
+    private lateinit var mapboxCameraTarget: com.mapbox.mapboxsdk.geometry.LatLng
+
+    /**
+     * initial setup
+     */
+    @Before
+    @Throws(Exception::class)
+    fun setUp() {
+        MockitoAnnotations.initMocks(this)
+        nearbyPresenter = Mockito.spy(NearbyParentFragmentPresenter(bookmarkLocationsDao))
```

Comment:
Do we need to be spying?

Reply:
Do you mean using a mock and "callRealMethod" instead?

Reply:
NearbyFragmentPresenter should be fine to be instantiated here, we don't need to wrap it in a spy I would think, it is the class under test